src/utils.cairo

```diff
@@ -1,34 +1,90 @@
-use core::traits::Into;
-use core::traits::TryInto;
+use core::num::traits::{Zero, One, BitSize};
+use core::starknet::secp256_trait::Secp256PointTrait;
 
-// Bitwise shift left for u256
-pub fn shl(value: u256, shift: u32) -> u256 {
-    value * fast_pow(2.into(), shift.into())
+pub trait Bitshift<T, U> {
+    fn shl(self: T, shift: U) -> T;
+    fn shr(self: T, shift: U) -> T;
 }
 
-// Bitwise shift right for u256
-pub fn shr(value: u256, shift: u32) -> u256 {
-    value / fast_pow(2.into(), shift.into())
+pub impl BitshiftImpl<
```

Comment:
Not sure if the trait is necessary. Why not keep it as a generic function?

Reply:
Thought It would be more elegant to have a trait for bitshift operation
I can remove it

Reply:
I don't think it is necessary.

Reply:
@maciejka  just updated the pr